### ISSUE TITLE
System editor issues - bodies not showing, file extension missing, etc

### ISSUE THREAD

**Issue by kerbalcmdr563** [pioneerspacesim/pioneer#5746]
EDIT: I'm adding further details, and additional issues to this report. (nozmajner)

Original issue:
When I create a new system, and only creating new system, any celestial body doesn't appear.

----------
### Bodies not shown
I can confirm this and add some details.
When creating a new system, nothing is visible, even if one adds bodies. Using the Center on Body command does nothing either.
![image](https://github.com/pioneerspacesim/pioneer/assets/4182678/90ebd52c-c7b0-4a02-b921-91d0a6cd5ae1)
Zooming does nothing either.

If an existing system is opened, it shows up properly, including the one one just saved.
![image](https://github.com/pioneerspacesim/pioneer/assets/4182678/41e24fa6-e42e-4995-ba28-6c579716c08a)

### Saves don't add file extension
In the save dialog, the .json extension isn't appended to the file name, and because of that, the saved system cannot be found.
(And I assume they won't be present in-game either.

### Tab cancels data entry
Pressing tab after entering data to any field does not accept the new value, but resets it to what it was before. Clicking away from the field does the same. I know that this might not be a straightforward fix, but from an UX perspective it would be preferable that both tab and clicking away would accept the new value. 

------
Win10, #5779 build,

**Comment by impaktor** [comment]
Is this reproducible? Could you please do a step by step description what you're doing to trigger the bug?

**Comment by Web-eWorks** [comment]
I'm unable to reproduce the missing save/open file dialogs on a Linux OS. I've reviewed the program-facing API I'm using to create the dialogs and as far as I can tell this isn't something I'm doing wrong when calling into the library. I'll look at the upstream dependency involved and see if the inability to open subsequent dialogs on Windows is tracked at their level and/or there's a fix I'm missing.

I'm able to (partially) reproduce the issue with TAB to enter values - it seems to work correctly on some entry widgets and fail on others. I'll dig further into that, but if you can provide a checklist of the entry widgets which are affected it'd be very helpful for validation purposes and save me some time.

**Comment by nozmajner** [comment]
Okay, I'll go through them

**Comment by Web-eWorks** [comment]
Hmm... I'd like to close this issue with the above PR as everything except the missing open/save dialogs are/will be fixed once merged.

@nozmajner could you break the missing editor open/save dialogs on Windows into a separate issue report? I'm not seeing anything obvious that would resolve that issue from the upstream dependency (https://github.com/samhocevar/portable-file-dialogs) but maybe @fluffyfreak could take a look at it and check it under a debugger?

**Comment by nozmajner** [comment]
Done
